subsys/bluetooth/mesh/brg_cfg.c

```diff
@@ -36,15 +36,19 @@ static ATOMIC_DEFINE(brg_cfg_flags, BRG_CFG_FLAGS_COUNT);
 static void brg_tbl_compact(void)
 {
 	int j = 0;
+	int k = 0;
 
-	for (int k = 0; k < bt_mesh_brg_cfg_row_cnt; k++) {
+	for (; k < bt_mesh_brg_cfg_row_cnt; k++) {
 		if (brg_tbl[k].direction != 0) {
 			brg_tbl[j] = brg_tbl[k];
 			j++;
 		}
```

Comment:
Could be good to add `j != k` to avoid extra copy. Though this won't be needed if starting index is passed to `brg_tbl_compact`.
```suggestion
		if (brg_tbl[k].direction != 0) {
			if (j != k) {
				brg_tbl[j] = brg_tbl[k];
			}
			j++;
		}
```